core/embed/projects/prodtest/cmd/prodtest_secrets.c

```diff
@@ -43,21 +49,26 @@
 secbool generate_random_secret(uint8_t* secret, size_t length) {
   random_buffer(secret, length);
 
+  uint8_t buffer[length];
 #ifdef USE_OPTIGA
-  uint8_t optiga_secret[length];
-  if (OPTIGA_SUCCESS != optiga_get_random(optiga_secret, length)) {
+  if (OPTIGA_SUCCESS != optiga_random_buffer(buffer, length)) {
     return secfalse;
   }
   for (size_t i = 0; i < length; i++) {
-    secret[i] ^= optiga_secret[i];
+    secret[i] ^= buffer[i];
   }
-  memzero(optiga_secret, sizeof(optiga_secret));
 #endif
 
 #ifdef USE_TROPIC
-  // TODO: Generate randomness using tropic and xor it with `secret`.
+  if (LT_OK != lt_random_value_get(tropic_get_handle(), buffer, length)) {
```

Comment:
I think this won't work in all cases because the secure channel with the tropic might not have been established yet.